dev/TreeView/TreeViewItemDataAutomationPeer.cpp

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "TreeView.h"
+#include "TreeViewItem.h"
+#include "TreeViewItemDataAutomationPeer.h"
+#include <UIAutomationCore.h>
+#include <UIAutomationCoreApi.h>
+
+CppWinRTActivatableClassWithBasicFactory(TreeViewItemDataAutomationPeer);
+
+TreeViewItemDataAutomationPeer::TreeViewItemDataAutomationPeer(winrt::TreeViewList owner, winrt::IInspectable const& item, winrt::ItemsControlAutomationPeer const& parent) :
+    ReferenceTracker(item, parent)
+{
+}
+
+// IExpandCollapseProvider 
+winrt::ExpandCollapseState TreeViewItemDataAutomationPeer::ExpandCollapseState()
+{
+    if (auto peer = GetTreeViewItemAutomationPeer())
+    {
+        return peer.ExpandCollapseState();
+    }
+    throw winrt::hresult_error(UIA_E_ELEMENTNOTENABLED);
+}
+
+void TreeViewItemDataAutomationPeer::Collapse()
+{
+    if (auto peer = GetTreeViewItemAutomationPeer())
+    {
+        return peer.Collapse();
+    }
+    throw winrt::hresult_error(UIA_E_ELEMENTNOTENABLED);
+}
+
+void TreeViewItemDataAutomationPeer::Expand()
+{
+    if (auto peer = GetTreeViewItemAutomationPeer())
+    {
+        return peer.Expand();
+    }
+    throw winrt::hresult_error(UIA_E_ELEMENTNOTENABLED);
+}
+
+// IAutomationPeerOverrides
+winrt::IInspectable TreeViewItemDataAutomationPeer::GetPatternCore(winrt::PatternInterface const& patternInterface)
+{
+    if (patternInterface == winrt::PatternInterface::ExpandCollapse)
+    {
+        return *this;
```

Comment:
Should we remove the ExpandCollapse pattern on TreeViewItemAutomationPeer?

Reply:
I think we should remove it from TreeViewItemAutomationPeer. Or does it have to be implemented on both peers to work correctly @licanhua ?

Reply:
@jevansaks What's your suggestion? 
We have these solutions:
1. Just TreeViewItemAutomationPeer as it is since narrator never use TreeViewItemAutomationPeer ExpandCollapse pattern in new implementation.
2. remove ExpandCollapse pattern from TreeViewItemAutomationPeer::GetPatternCore.
3. remove ExpandCollapse pattern from TreeViewItemAutomationPeer::GetPatternCore + remove the interface from idl.

Reply:
I will keep implemenation as it is.